src/internal/invoker/disposable.ts

```diff
@@ -0,0 +1,47 @@
+import { HandlerContext } from "../../HandlerContext";
+import { logger } from "../util/logger";
+
+export function registerDisposable(ctx: HandlerContext): (callback: () => Promise<void>, description?: string) => void {
```

Comment:
Why return a function that adds the disposable instead of just adding the disposable?

Reply:
That functions is being used when creating the context and gets assigned to `context.lifecycle.registerDisposable`. 

Since we don't have an actual implementation of `HandlerContext` that is way to park the implementation add it into the `HandlerContext` implementation when we create it.